_posts/2015-07-15-testautomation-allergy.md

```diff
@@ -0,0 +1,60 @@
+
+Of course there are different granularities in test automation. From unit-testing individual classes and methods to full blown user simulations. And they all have some importance to some task in developing software. Not always but often enough, people talk about test automation as a replacement for all other testing. They try to automate testing and then happily watch the machine do all the work. Sadly, test automation cannot perform the following check:
+
+> "Check that nothing happens that shouldn’t happen and everything else happens that should happen for all variations of this scenario and all possible states of systems and all possible visual/audio/tangible things user might observe and all possible states of the backend and all possible states of the system as a whole, and anything happening in the surrounding systems that should not matter but might matter."
+
+A human cannot test that on one go, but I bet my year’s salary that humans fare better at the task than a machine. Of course contexts vary and there might be occasions where a test automation can test absolutely everything. I (nor anyone whose opinion I value) have ever seen that.
+
+
+The framework of integrity and constant improvement
+----------------------------------------------------
+
+Why the f*** do we want to repeat a test? Why the f*** do we want to repeat a test? Why the f*** do we want to repeat a test? Why the f*** do we want to repeat a test? 
+
+See how silly it is? If you think about it, there are many variables we don't know. We have randomness in our servers, networks, everywhere! Even the bloody electrical current isn't stable So why on earth do we want to try and repeat the situation over and over, if the world we live in is not repeating itself? You might want to answer "to verify that regression doesn't exist". Isn't the point and purpose of regression testing to see if there is something broken? Shouldn't we be searching for the thing that MIGHT be broken?
+
+Actor Frank Langella said in a New York Times [article](http://www.nytimes.com/2014/01/05/theater/frank-langella-steps-into-king-lear-at-bam.html?hpw&rref=theater&_r=0)
+
+> “There’s a tendency to value consistency over creativity. You get it, you nail it, you repeat it. I’d rather hang myself. To me, every night within a certain framework — the framework of integrity — you must forget what you did the night before and create it anew every single time you walk out on the stage.”
+
+This is what test automation does: it finds a way that doesn't break the system and then repeats it. As a tester - as a thinking individual - I choose creativity and ideas before repeatability. I try not to do stupid things, but do the best possible testing every time I do testing. I don't fake it, I try to do the most important thing and I try to communicate my work to other people in a way that makes sense. Test automation doesn't strive for perfection, it settles for acceptable, meager existence.
```

Comment:
"This is what test automation does: it finds a way that doesn't break the system and then repeats it." Tää oli hyvä lause, tulee heti mieleen "wall of green" ja siitä saatava valheellinen hyvä mieli. Mulla oli vaikea ymmärtää mitä haet takaa ennen quotea olevilla kappaleilla, miettisin niitä vielä uusiksi.